### PR TITLE
[Performance] Improve memory usage and object retention

### PR DESCRIPTION
**Pull request by ahorek**
#### Why?
https://github.com/intercom/intercom-rails/pull/277

I run into the same issue as the original author.

#### How?
I rebased the original PR and fixed mentioned problems based on feedback.

### CODE REVIEW

**Comment by swrobel** [comment]
@jonnyom any hope of getting this reviewed & possibly merged or is this gem abandonware?

**Comment by swrobel** [comment]
@sduquej any hope of getting this reviewed & possibly merged or is this gem abandonware?

File: lib/intercom-rails/script_tag.rb [hunk 1]
```diff
@@ -73,12 +75,9 @@ def intercom_settings
...
     def to_s
-      js_options = 'id="IntercomSettingsScriptTag"'
-      if nonce && valid_nonce?
-        js_options = js_options + " nonce=\"#{nonce}\""
-      end
-      str = "<script #{js_options}>#{intercom_javascript}</script>\n"
-      str.respond_to?(:html_safe) ? str.html_safe : str
+      html_options = { id: 'IntercomSettingsScriptTag' }
+      html_options['nonce'] = nonce if valid_nonce?
+      javascript_tag intercom_javascript, html_options
```
Comment (sduquej) [inline lib/intercom-rails/script_tag.rb:80]:
```suggestion
      javascript_tag(intercom_javascript, html_options)
```

**Comment by sduquej** [comment]
This looks great, thanks @ahorek, @clowder (and @swrobel for the nudge)! I just have a couple of minor suggestions. 
I should have time this week to review it and cut a new release including these changes.

File: spec/auto_include_filter_spec.rb [hunk 2]
```diff
@@ -107,7 +107,7 @@ def current_user
     expect(response.body).to include(IntercomRails.config.app_id)
     expect(response.body).to include("ben@intercom.io")
     expect(response.body).to include("Ben McRedmond")
-    expect(response.body).to include("</script>\n</body>")
```
[the remark this answers is not in the document]
Comment (ahorek) [reply spec/auto_include_filter_spec.rb:110]: do you remember what was the intention to keep ```\n``` between these tags?

[the remark this answers is not in the document]
Comment (sduquej) [reply spec/auto_include_filter_spec.rb:110]: none other than to keep these in separate lines when looking at the source
```html
</script>
</body>
```

v.s. 
```html
</script></body>
```

[the remark this answers is not in the document]
Comment (ahorek) [reply spec/auto_include_filter_spec.rb]: ok, restored

File: lib/intercom-rails/auto_include_filter.rb [hunk 3]
```diff
@@ -33,7 +33,7 @@ def initialize(kontroller)
       end
 
       def include_javascript!
-        response.body = response.body.insert(response.body.index(CLOSING_BODY_TAG), intercom_script_tag.to_s)
+        response.body = response.body.insert(response.body.rindex(CLOSING_BODY_TAG), intercom_script_tag.to_s)
```
Comment (sduquej) [inline lib/intercom-rails/auto_include_filter.rb:36]: well thought! 👌

**Pull request merged**